### ISSUE TITLE
Add yoga layout support 

### ISSUE THREAD

**Issue by kusti8** [kusti8/proton-native#67]
I'd like to add yoga layout, because Grid is pretty buggy and it would make it more similar to React Native. There are a few things that need to be done to make this work. 

* [ ] Add fixed position support to libui (https://github.com/andlabs/libui/issues/306)
* [x] Add yoga-layout on the JS side
* [x] Create a wrapper called View over the libui Fixed widget

**Comment by kusti8** [comment]
I'm almost done with a version that works on Linux, I just need a few more days to work out the bugs. Once the proton-native side is ready, then there shouldn't be any more changes needed since everything else should be implemented in libui. 

For libui, the notable changes that I've made (on my fork) is add a fixed component using GtkFixed and add a way to get and set the size of any component. 

I need to find a way to implement that in Windows and Mac, but I assume they have easy support for fixed positioning and setting/getting the size. 

As for libui-node, I've added all of those bindings. Yoga isn't being built together with libui-node, but added with proton-native. I've had to change the compile options for libui-node because it and yoga were conflicting. 

The component is implemented as View and supports all of the options that View has on React Native. I need to see if I can implement setting the background color so that it's easier to see nested views.

**Comment by parro-it** [comment]
> . I've had to change the compile options for libui-node because it and yoga were conflicting.

Can you details where they conflict? And which options you need to. Change?

**Comment by kusti8** [comment]
I couldn't find exactly what conflicted, but I found this in nbind's issues and it worked. https://github.com/kusti8/libui-node/commit/57a957c4f23d89da53cfb141d96964eefc14bbf5

**Comment by kusti8** [comment]
I've added a PR to track my changes. All I need to do is add fixed positioning for Windows and Mac for libui and merge that.

**Comment by roman01la** [comment]
@kusti8 Is it still in development? I also have a question about running Yoga on JS side, it's interesting to see how it can impact perf 🤔

**Comment by mischnic** [comment]
@roman01la This PR needs to be merged first: https://github.com/andlabs/libui/pull/332.

**Comment by kusti8** [comment]
Proton Native V2 is now released! If the issue still occurs in the new update, please open a new issue.
